src/sync/remote_mongo_collection.cpp

```diff
@@ -18,3 +18,505 @@
 
 #include "sync/remote_mongo_collection.hpp"
 
+namespace realm {
+namespace app {
+
+static void handle_response(util::Optional<AppError> error,
+                            util::Optional<std::string> value,
+                            std::function<void(std::string, util::Optional<AppError>)> completion_block)
+{
+    if (value && !error) {
+        return completion_block(value.value(), error);
+    }
+    
+    return completion_block("", error);
+}
+
+static void handle_count_response(util::Optional<AppError> error,
+                                  util::Optional<std::string> value,
+                                  std::function<void(uint64_t, util::Optional<AppError>)> completion_block)
+{
+    if (value && !error) {
+        try {
+            auto json = nlohmann::json::parse(value.value());
+            auto count_string = json.at("$numberLong").get<std::string>();
```

Comment:
Can we do the int conversion here with `.get<uint64_t>()` instead of using `stoi`? I'd hope that the json conversion would be more robust than stoi, and would maybe enforce stricter json syntax. Plus if it fails, you'd get a json exception (hopefully more descriptive), rather than a generic "invalid argument".

Reply:
issue is the number is returned as a string e.g. `"{\"$numberLong\":\"19\"}"` so a parse from string to int need to be performed